Chat-spaceに非同期通信を実装

y0913 opened the pull request:
Chat-spaceに非同期通信を実装しました。
ご確認お願い致します!!

Commit:
Chat-spaceに非同期通信を実装

Chat-spaceに非同期通信を実装しました。
ご確認お願い致します!!

kuroko918 commented:
コミットは一度にすべてするのではなく、ファイルごとやアクションごとなど、もう少し細かくコミットしていきましょう！

kuroko918 commented on app/controllers/messages_controller.rb:
```diff
@@ -8,8 +8,14 @@ def index
+
     if @message.save
-      redirect_to group_messages_path(@group), notice: 'メッセージが送信されました'
+      # redirect_to group_messages_path(@group), notice: 'メッセージが送信されました'
```
不要なコメントは消しましょう！

kuroko918 commented on app/assets/javascripts/comments.js:
```diff
@@ -0,0 +1,46 @@
+		})
+		.fail(function(){
+			alert('error');
+			$('.form__submit').prop('disabled', false);
```
この記述はdoneの場合にもあるのでalwaysを使ってまとめましょう！

kuroko918 commented on app/assets/javascripts/comments.js:
```diff
@@ -0,0 +1,46 @@
+					</div>
+					<div class="main-content__message-body">
+						${message.content}
+						<img src="${message.image}">
```
imageがない場合は、どのようになっていますでしょうか？
imageがない場合は何も表示されないように変更しましょう！
messageImageなどの変数を定義し、その変数に三項演算子を使用して、imageがある場合とない場合のimgタグを代入しましょう！

Commit:
Edit comments.js & messages_controller.rb

Messagesコントローラーにて不要なコメントを削除しました。
Comments.js にてprop('disabled', false)を削除しました。（動作確認して不要だったためです。）
イメージがない場合の記述も追加いたしました。

Commit:
Edit comment.js

$('.form__submit').prop('disabled', false);
上記一文を削除いたしました。

yutakatahira commented on app/assets/javascripts/comments.js:
```diff
@@ -0,0 +1,46 @@
+$(function(){
+	function buildHTML(message){
+		var addImage = (message.image !== null)? `<img src = "${message.image}">`:''
+	var html = `<div class="main-content__message-chat">
```
同じような条件分岐をメッセージで行ってみましょう！

ky3090 commented on app/assets/javascripts/comments.js:
```diff
@@ -0,0 +1,46 @@
+			var html = buildHTML(data);
+			$('.main-content__message').append(html);
+			$('.form__submit').prop('disabled', false);
+			$('.form__message').val('');
```
フォームが空の状態で、またSENDボタンを押すと、前回のメッセージが再度送れませんか？

yutakatahira commented on app/assets/javascripts/comments.js:
```diff
@@ -0,0 +1,46 @@
+$(function(){
+	function buildHTML(message){
+		var addImage = (message.image !== null)? `<img src = "${message.image}">`:''
```
同じような条件分岐をメッセージでも行ってみましょう！

Commit:
Edit comments.js

Comments.jsを編集しました。
よろしくお願いいたします。

kohei-isoda commented on app/assets/javascripts/comments.js:
```diff
@@ -0,0 +1,48 @@
+	};
+
+	function scroll(){
+		$('.messages').animate({scrollTop: $('.message'[0].scrllHeight)});
```
スペルミスがございますので修正お願いします！

kohei-isoda commented on app/assets/javascripts/comments.js:
```diff
@@ -0,0 +1,48 @@
+$(function(){
+	function buildHTML(message){
+		var addImage = (message.image !== null)? `<img src = "${message.image}">`:''
+		var content = (message.content !== null)? `${message.content}`:''
```
こちら不要ですので削除しましょう！
（あってもなくても変わりませんと言う意味合いになります）

kohei-isoda commented on app/views/messages/create.json.jbuilder:
```diff
@@ -0,0 +1,5 @@
+json.user_name @message.user.name
+json.image @message.image.url
+json.content @message.content
+json.id @message.id
```
【質問】
idはjbuilderでは使用していないみたいなのですが、使用する予定があるということでしょうか？

y0913 replied:
ご確認ありがとうございます。
使用する予定がありませんので削除いたします。

よろしくお願いいたします。

kohei-isoda commented on app/assets/javascripts/comments.js:
```diff
@@ -0,0 +1,48 @@
+		.done(function(data){
+			var html = buildHTML(data);
+			$('.main-content__message').append(html);
+			$('.form__submit').prop('disabled', false);
```
failにも必要な処理かと思います！

Commit:
Edit comments.js

不要部分を削除しました。

Commit:
Edit comment.js

var addImage = (message.image !== null)? `<img src =
"${message.image}">`:''
上記を追加

.prop(‘disabled, false); をalwaysにまとめて追加しました。

kohei-isoda commented on app/assets/javascripts/comments.js:
```diff
@@ -0,0 +1,50 @@
+$(function(){
+	function buildHTML(message){
+		var addImage = (message.image !== null)? `<img src = "${message.image}">`:''
```
nullでなければという条件式は基本的には避けたいですね！

y0913 replied:
ご確認ありがとうございます。

!==を== に修正し、対応いたしました。

Commit:
EDIT comments.js

三項演算子の一文を書き換えました。
